pkg/chains/storage/docdb/docdb.go

```diff
@@ -125,3 +252,77 @@ func (b *Backend) retrieveDocuments(ctx context.Context, opts config.StorageOpts
 	}
 	return []SignedDocument{d}, nil
 }
+
+func populateMongoServerURL(ctx context.Context, cfg config.Config) error {
+	// First preference is given to the key `storage.docdb.mongo-server-url-dir`.
+	// If that doesn't exist, then we move on to `storage.docdb.mongo-server-url`.
+	// If that doesn't exist, then we check if `MONGO_SERVER_URL` env var is set.
+	logger := logging.FromContext(ctx)
+	mongoEnv := "MONGO_SERVER_URL"
+
+	if cfg.Storage.DocDB.MongoServerURLDir != "" {
+		logger.Infof("setting %s from storage.docdb.mongo-server-url-dir: %s", mongoEnv, cfg.Storage.DocDB.MongoServerURLDir)
+		if err := setMongoServerURLFromDir(cfg.Storage.DocDB.MongoServerURLDir); err != nil {
+			return err
+		}
+	} else if cfg.Storage.DocDB.MongoServerURL != "" {
+		logger.Infof("setting %s from storage.docdb.mongo-server-url", mongoEnv)
+		if err := os.Setenv(mongoEnv, cfg.Storage.DocDB.MongoServerURL); err != nil {
```

Comment:
> I'm not a fan of setting global env state from the local config

makes sense and i agree, however there 2 things here:
- `os.Setenv` will only set the value of `MONGO_SERVER_URL` within this context in code and not outside for the whole container and it can't update its parent process
- we must set `MONGO_SERVER_URL` env var instead of a variable in config because that's what https://github.com/google/go-cloud/tree/master/docstore/mongodocstore consumes...

> I'd also expect this ordering for config values, since they're most specific to least.
ConfigMap values set directly
MONGO_SERVER_URL
MONGO_SERVER_URL_DIR

- this is roughly the order that's set right now; ConfigMap values take precedence over `MONGO_SERVER_URL`
- `MONGO_SERVER_URL_DIR` is not an env var that i've added 🤔